### ISSUE TITLE
Variable defined in while condition incorrectly marked as PossiblyUndefinedGlobalVariable

### ISSUE THREAD

**Issue by wouterj** [vimeo/psalm#5676]
https://psalm.dev/r/9756964e29

The condition is always run at least once, so we can be sure that the variable exists after it.

**Comment by muglug** [comment]
Simplified to

```php
<?php

while ($r = rand(0, 1)) {}
echo $r;
```
